web/src/engine/osk/src/banner/banner.ts

```diff
@@ -342,34 +363,119 @@ export class SuggestionBanner extends Banner {
     }
   }
 
-  private setupInputHandling() {
-    let inputEngine: InputEventEngine;
-    if(this.hostDevice.touchable) { //  /*&& ('ontouchstart' in window)*/ // Except Chrome emulation doesn't set this.
-      // Not to mention, it's rather redundant.
-      inputEngine = this.touchEventConfig;
-    } else {
-      inputEngine = this.mouseEventConfig;
+  private setupInputHandling(): GestureRecognizer<BannerSuggestion> {
+
+    const findTargetFrom = (e: HTMLElement): HTMLDivElement => {
+    try {
+      if(e) {
+        if(e.classList.contains('kmw-suggest-option')) {
+          return e as HTMLDivElement;
+        }
+        if(e.parentElement && e.parentElement.classList.contains('kmw-suggest-option')) {
+          return e.parentElement as HTMLDivElement;
+        }
+      }
+    } catch(ex) {}
+      return null;
     }
 
-    inputEngine.registerEventHandlers();
+    const config: GestureRecognizerConfiguration<BannerSuggestion> = {
+      targetRoot: this.getDiv(),
+      maxRoamingBounds: new PaddedZoneSource(this.getDiv(), [-0.333 * this.height]),
+      // touchEventRoot:  this.element, // is the default
+      itemIdentifier: (sample, target: HTMLElement) => {
+        let bestMatch: BannerSuggestion = null;
+        let bestDist = Number.MAX_VALUE;
+
+        for(let i = 0; i < this.options.length; i++) {
+          const option = this.options[i];
```

Comment:
```suggestion
        for(const option of this.options) {
```

Seems to be an option?